prov/gni/src/gnix_rma.c

```diff
@@ -1452,14 +1455,13 @@ ssize_t _gnix_rma(struct gnix_fid_ep *ep, enum gnix_fab_req_type fr_type,
 			slist_insert_tail(&req->rma.sle, &ep->more_read);
 			req->work_fn = _gnix_rma_more_post_req;
 		}
-		if (flags & FI_MORE)
-			return FI_SUCCESS;
+
+		/* Req is now in a chain, do not queue it alone. */
+		req = NULL;
```

Comment:
The request can be queued to be sent alone or as part of a chain.  This code is inside the block of code where the request was just added to  a chain.  So, it should not be queued by itself, alone.